journalsservice/views.py

```diff
@@ -151,3 +151,35 @@ def get(self, bibstem):
                         'Error Info': 'Unspecified error.  Try again.'}, 500
         return {'refsource': request_json}, 200
 
+
+class ISSN(Resource):
+
+    scopes = []
+    rate_limit = [1000, 60 * 60 * 24]
+    decorators = [advertise('scopes', 'rate_limit')]
+
+    def get(self, issn):
+        request_json = {}
+        if issn:
+            try:
+                if len(issn) == 8:
+                    issn = issn[0:4] + "-" + issn[4:]
+                with current_app.session_scope() as session:
+                    dat_idents = session.query(JournalsIdentifiers).filter(and_(JournalsIdentifiers.id_value==issn, JournalsIdentifiers.id_type.like("ISSN%"))).first()
+                    if dat_idents:
+                        masterid = dat_idents.masterid
+                        id_value = dat_idents.id_value
+                        id_type = dat_idents.id_type
+                        dat_master = session.query(JournalsMaster).filter_by(masterid=masterid).first()
+                        bibstem = dat_master.bibstem
+                        journal_name = dat_master.journal_name
+                        request_json = {'issn': {'ISSN': id_value,
+                                                'ISSN_type': id_type,
+                                                'bibstem': bibstem,
+                                                'journal_name': journal_name}}
+            except Exception as err:
+                return {'Error': 'Refsource search failed',
```

Comment:
You should include "err" in the information you return when the exception occurs. It may contain useful information.